confdgnmi/src/confd_gnmi_api_adapter.py

```diff
@@ -59,32 +63,183 @@ def get_inst(cls) -> GnmiServerAdapter:
+            self.change_db_lock = threading.Lock()
+            self.change_thread = None
+            self.stop_pipe = None
+
+        def get_monitored_changes(self) -> []:
+            # TODO reuse with demo adapter ?
+            self.change_db_lock.acquire()
+            log.debug("==> self.change_db=%s", self.change_db)
+            assert len(self.change_db) > 0
+            update = []
+            for c in self.change_db:
+                prefix_str = make_xpath_path(
+                    gnmi_prefix=self.subscription_list.prefix)
+                p = c[0]
+                if c[0].startswith(prefix_str):
+                    p = c[0][len(prefix_str):]
+                update.append(gnmi_pb2.Update(path=make_gnmi_path(p),
+                                              val=gnmi_pb2.TypedValue(
+                                                  string_val=c[1])))
+            self.change_db = []
+            self.change_db_lock.release()
+            log.debug("<== update=%s", update)
+            return update
+
+        def get_sample(self, path, prefix,
+                       start_change_processing=False):
+            update = []
             log.debug("==>")
-            assert self.subscription_list != None
-            # for now we only process GET type (fetch and return everything)
-            assert self.subscription_stream_event_type == self.SubscriptionStreamEventType.GET
+            update.extend(self.adapter.get_updates_with_maapi_save(path,
+                                                                   prefix,
+                                                                   gnmi_pb2.GetRequest.DataType.ALL))
+            log.debug("<== update=%s", update)
+            return update
 
-            update = []
-            for s in self.subscription_list.subscription:
-                update.extend(self.adapter.get_updates_with_maapi_save(s.path,
-                                                                       self.subscription_list.prefix,
-                                                                       gnmi_pb2.GetRequest.DataType.ALL))
-            # TODO delete
-            delete = []
-            notif = gnmi_pb2.Notification(timestamp=0,
-                                          prefix=self.subscription_list.prefix,
-                                          alias="/alias", update=update,
-                                          delete=delete,
-                                          atomic=False)
-
-            response = gnmi_pb2.SubscribeResponse(update=notif)
-            log.debug("<== response=%s", response)
-            return response
-
-    def get_subscription_handler(self) -> SubscriptionHandler:
+        def add_path_for_monitoring(self, path, prefix):
+            log.debug("==>")
+            path_with_prefix_str = make_formatted_path(path, prefix)
+            self.monitored_paths.append(path_with_prefix_str)
+            log.debug("<==")
+
+        def kp_to_xpath(self, kp):
```

Comment:
The code here is a bit too frail, I think (such as: there can be at most one prefix in the path).

Reply:
Yes, it is TODO (I'll update coment)